### PR TITLE
LPS-146573

### PR DESCRIPTION
**Pull request by david-truong**

### CODE REVIEW

**Comment by xbrianlee** [comment]
@suilin analyzed the PR. The results look good and no issues are related the changes.

We're continuing to work on adding functional tests for these changes in [LRQA-75196](https://issues.liferay.com/browse/LRQA-75196) however this will take a bit more time. Feel free to forward this PR without tests. We'll send the tests up soon after.

**Comment by suilin** [comment]
Hi @david-truong, I'm working on adding the functional tests, but I find there is a difference from acceptance criteria that Sam mentioned before:

> * When sandbox mode is enabled when previously disabled, all users that apply will be sandboxed.

Test steps:
* Login in with test user
* Enable Sandbox only
* Add a new Publication
* Add some contents
* Delete/Publish the Publication

Result: There is no sandbox publication for test user, I'd like to confirm with you is it the expected result?

**Comment by david-truong** [comment]
@suilin you are correct

1. we switched to screen name because many people have the same first and last name but screen name is unique in liferay.
2. one thing we did change is that a user will only get sandboxed the next time they login or publish something.  we don't sandbox them immediately after enabling sandbox.  test user is not sandboxed because he has the work on production permission.

**Comment by suilin** [comment]
@david-truong, thanks for your confirmation.

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/events/LoginPostAction.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+			if (!_ctSettingsConfigurationHelper.isEnabled(
+					user.getCompanyId()) ||
+				!_ctSettingsConfigurationHelper.isSandboxEnabled(
+					user.getCompanyId())) {
+
+				return;
+			}
+
+			PermissionChecker permissionChecker =
```
Comment (david-truong) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/events/LoginPostAction.java:68]: @tinatian themeDisplay was always null so I couldn't use it.  Is this okay?

**Comment by david-truong** [comment]
ci:test:sf

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:test:publications

**Comment by david-truong** [comment]
ci:stop

**Comment by david-truong** [comment]
ci:test:sf

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:test:publications

**Comment by suilin** [comment]
Hi @david-truong, 
I anayzed the latest results, they're look good.
And I finished adding functional tests for these changes in [LRQA-75196](https://issues.liferay.com/browse/LRQA-75196), and I also combine the pr and tests on https://github.com/suilin/liferay-portal/pull/529, the tests works fine. Do you think we can forward them? Thanks.

cc @xbrianlee

**Comment by david-truong** [comment]
https://github.com/liferay-publications/liferay-portal/pull/124

**Pull request closed without merging**